### ISSUE TITLE
Support Clojure / deps.edn projects

### ISSUE THREAD

**Issue by devurandom** [opencollective/backyourstack#97]
It would be great if BackYourStack could support [Clojure](https://clojure.org) projects that use [deps.edn](https://clojure.org/guides/deps_and_cli#_running_a_repl_and_using_libraries).

The file you want to be looking at is a Clojure / [EDN](https://github.com/edn-format/edn) file, the same format as relevant for #96. A sample that should be sufficient for the purposes of BackYourStack is given in the [deps and CLI guide](https://clojure.org/guides/deps_and_cli#_running_a_repl_and_using_libraries), with a full reference / extensive documentation being available in the Clojure reference sections about [deps.edn](https://clojure.org/reference/deps_and_cli#_deps_edn) and [dependencies](https://clojure.org/reference/deps_and_cli#_dependencies). These files contain exactly one map `{}` of `:keyword` value pairs. You are searching for the value associated with the `:deps` keyword, which in turn will be a map `{}` of `symbol` value pairs, with the value again being a map `{}` of `:keyword` value pairs. In this final map, the only keyword interesting to you is `:mvn/version` whose value will be a `"string"`.

Sample taken from the [Clojure reference on dependencies](https://clojure.org/reference/deps_and_cli#_dependencies):
```edn
{:deps
 {org.clojure/tools.reader {:mvn/version "1.1.1"}}}
```

The dependencies with their descriptions, home page, source repo, versions, dependencies, licenses are usually to be found on [Clojars](https://clojars.org/) ([HTTP REST API documentation](https://github.com/clojars/clojars-web/wiki/Data)) and in some rare cases on [Maven](https://maven.org/). This is the same as for #96.

**Comment by martinklepsch** [comment]
I think the easiest and most reliable way to deal with this is by looking at projects `pom.xml`. This means an implementation of this does not have to deal with build tool specific files and the implementation will likely be very similar to a Java implementation.

Is there a guide to implement support for new languages? What is needed, etc? 

This also applies to #96 so maybe these issues can be considered duplicates.

**Comment by znarf** [comment]
@martinklepsch If `pom.xml` files are committed to the repositories, it does sound like a reasonable implementation. Can you point to example repositories that use Clojure and have a `pom.xml` file?

We don't have a guide yet to support new languages. But:
 - You can look at current implementations there: https://github.com/opencollective/backyourstack/tree/master/src/lib/dependencies
 - There is a _template` to help you get started: https://github.com/opencollective/backyourstack/blob/master/src/lib/dependencies/_template.js
 - You can also look at past issues for Nuget #56, Ruby #60, Go #75

**Comment by devurandom** [comment]
@martinklepsch Do you plan on implementing this or are you merely gathering information for the one who will?

Re: `pom.xml`: I see how they can be used to learn about the dependencies' dependencies, but we are not guaranteed that also the initial project is available on Clojars. Thus I think there is no way around parsing EDN.

**Comment by martinklepsch** [comment]
@devurandom you’re right, I assumed you would run BYS on a dependency but really most people are going to run it on their project that may not even have a pom.xml or jar packaging.

With that in mind we’ll need something that 

1. parses some local project file (`project.clj`, `build.boot`, `deps.edn`) to get the set of dependencies used by the application / project
2. retrieves dependencies of those dependencies (i.e. transitive dependencies) 

Just for my understanding, does backyourstack mostly run on the users computer or on a server in some way? (On mobile right now so can’t look it up easily but don’t worry if there’s no quick answer.)

**Comment by devurandom** [comment]
Until now I only knew about https://backyourstack.com/ i.e. examining people's stacks online -- I did not even know you could (conveniently) run it on your local computer.

**Comment by znarf** [comment]
As of today, BackYourStack is effectively only a website that is available at https://backyourstack.com/

It does make sense to make it available as a CLI too, but that's absolutely not there. Would be happy to help if someone wants to work on that.

**Comment by martinklepsch** [comment]
Ah, I think I confused it with @feross’ `thanks` project, sorry 😊
